lib/CMakeLists.txt

```diff
@@ -203,9 +206,9 @@ include(CheckLinkerFlag)
 check_linker_flag(CXX "-Wl,--compress-debug-sections=zlib" QUDA_LINKER_COMPRESS)
 target_link_options(quda PRIVATE $<$<CONFIG:DEVEL>:$<${QUDA_LINKER_COMPRESS}:-Wl,--compress-debug-sections=zlib>>)
 
-# set up QUDA compile options
+# set up QUDA compile options, put them before target specific options
 target_compile_options(
-  quda PRIVATE $<$<COMPILE_LANGUAGE:CXX,C,Fortran>: 
+  quda BEFORE PRIVATE $<$<COMPILE_LANGUAGE:CXX,C,Fortran>:
```

Comment:
just curious: Is this style or are there technical reasons with sycl?

Reply:
It was a technical reason.  Some warnings need to be disabled on SYCL at present, and so any target-specific warning disablement has to be done after the general warning are enabled.